File: algorithms/easy/find_the_maximum_divisibility_score.ex
defmodule Solution do
  @spec max_div_score(nums :: [integer], divisors :: [integer]) :: integer
  def max_div_score(nums, divisors) do
    divisors
    |> Enum.map(fn d ->
      {Enum.count(nums, &(rem(&1, d) == 0)), d}
    end)
    |> Enum.max_by(fn {cnt, d} -> {cnt, -d} end)
    |> elem(1)
  end
end

File: algorithms/easy/find_the_maximum_divisibility_score.rs
impl Solution {
    pub fn max_div_score(nums: Vec<i32>, divisors: Vec<i32>) -> i32 {
        let mut ans = divisors[0];
        let mut max = 0;
        for d in divisors {
            let mut ct = 0;
            for n in &nums {
                if n % d == 0 {
                    ct += 1;
                }
            }
            if max < ct {
                max = ct;
                ans = d;
            } else if max == ct {
                ans = ans.min(d);
            }
        }
        ans
    }
}
